python/rbac/ldap.py

```diff
@@ -98,40 +127,36 @@ def _get_ldap_default_settings():
             'CACHE_TIMEOUT': 3600,
         }
 
-        if 'ldaps://' in ldap_config['ldap_uri'].lower():
+        if is_tls(ldap_config['ldap_uri']):
             cert_filepath = ldap_config.get('tls_ca_cert_file', '')
             if not cert_filepath or not os.path.exists(cert_filepath):
-                raise AdcmEx('LDAP_NO_CERT_FILE')
-
-            connection_options = {
-                ldap.OPT_X_TLS_CACERTFILE: cert_filepath,
-                ldap.OPT_X_TLS_REQUIRE_CERT: ldap.OPT_X_TLS_ALLOW,
-                ldap.OPT_X_TLS_NEWCTX: 0,
-            }
+                log.warning('missing cert file for `ldaps://` connection')
+                return {}
+            connection_options = configure_tls(enabled=True, cert_filepath=cert_filepath)
             default_settings.update({'CONNECTION_OPTIONS': connection_options})
-            os.environ['CERT_ENV_KEY'] = cert_filepath
 
         return default_settings
 
     return {}
 
 
 class CustomLDAPBackend(LDAPBackend):
+    default_settings = {}
+    is_tls = False
+
     def authenticate_ldap_user(self, ldap_user, password):
         self.default_settings = _get_ldap_default_settings()
         if not self.default_settings:
             return None
+        self.is_tls = is_tls(self.default_settings['SERVER_URI'])
```

Comment:
self property needs to be set in __init__ as well as default_settings and is_tls

Reply:
fixed in https://github.com/arenadata/adcm/pull/1874